vault/core_metrics_test.go

```diff
@@ -0,0 +1,111 @@
+package vault
+
+import (
+	"testing"
+
+	"github.com/hashicorp/vault/helper/namespace"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func TestCoreMetrics_KvSecretGauge(t *testing.T) {
+	core, _, root := TestCoreUnsealed(t)
+
+	// I think I can't test with the real kv-v2 because it's
+	// a plugin. But we can fake it by using "metadata" as
```

Comment:
It's a builtin plugin so you should be able to test with kvv2 - plenty of existing tests do.

Reply:
What do I need to do to make that happen? Can you give me a pointer?

One example I found was in external_tests which explicitly linked the plugin package, so I guess that would work. The others I found were in the command package which starts a test server, which seems inappropriate here.

Reply:
Shouldn't need more than

```
	if err := testClient.Sys().Mount("kv", &api.MountInput{
		Type: "kv",
		Options: map[string]string{
			"version": "2",
		},
	}); err != nil {
		t.Fatal(err)
	}
```

Reply:
Or look at `testCoreAddSecretMount`

Reply:
OK; I admit I don't understand why going directly to core.mount would behave any differently than going through HandleRequest, but I can give it a try.

I thought the problem is that the test configuration uses
        logicalBackends["kv"] = LeasedPassthroughBackendFactory
so we always get v1.

Reply:
Yeah, changing how I mounted the secret engine isn't the problem; it still goes to PassthroughBackend. I think I need to replace core's configuration with something like:

```
	coreConfig := &vault.CoreConfig{
		LogicalBackends: map[string]logical.Factory{
			"kv": logicalKv.Factory,
		},
	}
```

Reply:
I managed to make this work, but using a mechanism it looks like nobody else ever has (for swapping in a different backend factory into TestCore.)